pkg/serverless/logs/http.go

```diff
@@ -0,0 +1,32 @@
+package logs
+
+import (
+	"io/ioutil"
+	"net/http"
+)
+
+// LambdaLogsAPI implements the AWS Lambda Logs API callback
+type LambdaLogsAPIServer struct {
+	out chan<- []LambdaLogAPIMessage
+}
+
+func NewLambdaLogsAPIServer(out chan<- []LambdaLogAPIMessage) LambdaLogsAPIServer {
+	return LambdaLogsAPIServer{out}
+}
+
+func (l *LambdaLogsAPIServer) Close() {
+	close(l.out)
+}
+
+// ServeHTTP - see type LambdaLogsCollector comment.
+func (c *LambdaLogsAPIServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	data, _ := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	messages, err := parseLogsAPIPayload(data)
```

Comment:
I'm wondering if we should move `parseLogsAPIPayload` to the other side of the channel as well? What are your thoughts on this?

Reply:
The API is meant to determine whether a valid payload has been received. If there was a formatting issue, and we deferred unmarshalling until after we have returned a 200, the issue would be obfuscated.